### PR TITLE
Nfc sample docs

### PR DESCRIPTION
**Pull request by kapi-no**
Adding missing documentation for NFC samples

### CODE REVIEW

**Comment by kapi-no** [comment]
@ru-fu, @grochu, please review this PR

File: samples/nfc/writable_ndef_msg/README.rst [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+Testing
+=======
+
+After programming the sample to your board, test it by performing the following steps:
+
+1. Touch the NFC antenna with the smartphone or tablet and observe that LED 1 and LED 4 are lit.
+#. Observe that the smartphone/tablet tries to open the URL "http\://www.nordicsemi.com" in a web browser.
+#. Use a proper application (for example, NFC Tools for Android) to overwrite the existing NDEF message with your own message.
+#. Power-cycle your board and touch the antenna again.
+   Observe that the new message is displayed.
```
Comment (ru-fu) [inline samples/nfc/writable_ndef_msg/README.rst]: I added this step - does this make sense?

File: include/nfc/ndef/nfc_ndef.rst [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+You must provide header information for the record (for example, ID and type) and the payload data to the macro.
+For generic data that is not in binary format, you must also provide a payload constructor.
+This constructor converts the payload data to the needed format.
+For binary data, the module provides this constructor.
+
+The following code example shows how to generate a record descriptor:
+
+.. code-block:: c
+
+   ret_code_t error_code;
```
Comment (ru-fu) [inline include/nfc/ndef/nfc_ndef.rst]: I have not updated this code block and the one below - please check.

File: include/nfc/ndef/nfc_text.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+Text records contain a payload (the actual text string) and a language identifier.
+Note that a message can contain only one text record per language code, so you cannot add two different text records with language code "en" to the same message, for example.
+
+The following code example shows how to generate a raw text message that contains a text record.
+
+First, generate the text record.
+In this example, the record contains the text "Hello World!" and the language code "en":
+
+.. code-block:: c
+
```
Comment (ru-fu) [inline include/nfc/ndef/nfc_text.rst]: I copied the code blocks in this file from the sample - but please check.

File: include/nfc/ndef/nfc_uri.rst [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+See :cpp:enum:`nfc_uri_id` for the available protocols.
+
+The :ref:`nfc_uri_record` module provides functions for creating the record, and the :ref:`nfc_uri_msg` module provides functions for creating and encoding the message.
+
+The following code snippets show how to generate a URI message.
+First, define the URI string and create a buffer for the message:
+
+.. code-block:: c
+
+   static const u8_t m_url[] =
```
Comment (ru-fu) [inline include/nfc/ndef/nfc_uri.rst]: I've tried to update the code blocks in this file - but please check if they are correct.

File: samples/nfc/writable_ndef_msg/README.rst (hunk 1 above)
Comment (kapi-no) [reply samples/nfc/writable_ndef_msg/README.rst]: yes, let's keep it

File: include/nfc/ndef/nfc_text.rst (hunk 3 above)
Comment (kapi-no) [reply include/nfc/ndef/nfc_text.rst]: I did and it looks alright. Should I remove the snippets from the sample code if they aren't used? Or are we going to introduce at some point similar snippet-like functionality that we used in the previous doc framework?

File: include/nfc/ndef/nfc_uri.rst [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                                   m_url,
+                                   sizeof(m_url),
+                                   m_ndef_msg_buf,
+                                   &len);
+
+   if (err < 0) {
+        printk("Cannot encode message!\n");
+   }
+
+   return err;
```
Comment (kapi-no) [inline include/nfc/ndef/nfc_uri.rst]: This line can be removed

File: include/nfc/ndef/nfc_uri.rst (hunk 4 above)
Comment (kapi-no) [reply include/nfc/ndef/nfc_uri.rst]: One small fix - otherwise looks ok

File: include/nfc/ndef/nfc_ndef.rst (hunk 2 above)
Comment (kapi-no) [reply include/nfc/ndef/nfc_ndef.rst]: a few comments for this one to make it more Zephyr like

Comment (kapi-no) [inline include/nfc/ndef/nfc_ndef.rst]: ret_code_t -> int
error_code -> err (and all further occurences)

File: include/nfc/ndef/nfc_ndef.rst [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+   uint32_t length;
```
Comment (kapi-no) [inline include/nfc/ndef/nfc_ndef.rst]: uint32_t -> u32_t

File: include/nfc/ndef/nfc_ndef.rst [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+:cpp:func:`nfc_ndef_msg_encode` internally calls :cpp:func:`nfc_ndef_record_encode` to encode each record.
+The NDEF records are always encoded in long format.
+If no ID field is specified, a record without ID field is generated.
+
+The following code example shows how to create two messages:
+
+
+.. code-block:: c
+
+   ret_code_t error_code;
```
Comment (kapi-no) [inline include/nfc/ndef/nfc_ndef.rst]: ret_code_t  -> int
error_code -> err

File: include/nfc/ndef/nfc_ndef.rst [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+   uint8_t buffer_for_message[512];
```
Comment (kapi-no) [inline include/nfc/ndef/nfc_ndef.rst]: uint8_t  -> u8_t

File: include/nfc/ndef/nfc_ndef.rst [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+   uint8_t buffer_for_message_2[128];
```
Comment (kapi-no) [inline include/nfc/ndef/nfc_ndef.rst]: ditto

File: include/nfc/ndef/nfc_ndef.rst [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+   uint32_t length;
```
Comment (kapi-no) [inline include/nfc/ndef/nfc_ndef.rst]: uint32_t -> u32_t

File: include/nfc/ndef/nfc_text.rst [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+   err = nfc_ndef_msg_encode(&NFC_NDEF_MSG(nfc_text_msg),
+				         buffer,
+				         len);
+   if (err < 0) {
+	printk("Cannot encode message!\n");
+   }
+
+   return err;
```
Comment (kapi-no) [inline include/nfc/ndef/nfc_text.rst]: please remove it

File: include/nfc/ndef/nfc_text.rst [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+
+Finally, encode the message:
+
+.. code-block:: c
+
+   err = nfc_ndef_msg_encode(&NFC_NDEF_MSG(nfc_text_msg),
+				         buffer,
+				         len);
+   if (err < 0) {
+	printk("Cannot encode message!\n");
```
Comment (kapi-no) [inline include/nfc/ndef/nfc_text.rst]: please add
```return err;```
after this line

File: include/nfc/ndef/nfc_uri.rst [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+   int err;
+
+   err = nfc_uri_msg_encode( NFC_URI_HTTP_WWW,
+                                   m_url,
+                                   sizeof(m_url),
+                                   m_ndef_msg_buf,
+                                   &len);
+
+   if (err < 0) {
+        printk("Cannot encode message!\n");
```
Comment (kapi-no) [inline include/nfc/ndef/nfc_uri.rst]: please add
```return err;```
after this line

File: include/nfc/ndef/nfc_text.rst (hunk 3 above)
Comment (ru-fu) [reply include/nfc/ndef/nfc_text.rst]: I don't think there are code snippets like that in RST - @carlescufi ?

**Comment by ru-fu** [comment]
Thanks - updated according to the comments.
Any idea why the CI is failing?

File: include/nfc/ndef/nfc_text.rst (hunk 3 above)
Comment (carlescufi) [reply include/nfc/ndef/nfc_text.rst]: @ru-fu I am not sure what you mean by "like that"? maybe you can point me to an example?

Comment (carlescufi) [reply include/nfc/ndef/nfc_text.rst]: You can include source code directly from another file if that is what you mean, take a look at how to do it here:

```
.. literalinclude:: ../../../samples/bluetooth/beacon/src/main.c
   :language: c
   :lines: 19-
   :linenos:
```

https://raw.githubusercontent.com/zephyrproject-rtos/zephyr/master/doc/guides/bluetooth/index.rst

**Pull request merged**